aquadoggo/src/replication/announcement.rs

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use p2panda_rs::Validate;
+use serde::de::Visitor;
+use serde::ser::SerializeSeq;
+use serde::{Deserialize, Serialize};
+
+use crate::replication::{MessageType, TargetSet, ANNOUNCE_TYPE, REPLICATION_PROTOCOL_VERSION};
+
+/// u64 timestamp from UNIX epoch until now.
+pub fn now() -> u64 {
+    SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .expect("System time invalid, operation system time configured before UNIX epoch")
+        .as_secs()
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Announcement {
+    /// This contains a list of schema ids this peer is interested in.
+    pub target_set: TargetSet,
```

Comment:
I wonder if it would be better to use different naming here rather than "target_set" which is maybe quite a low-level term which comes from the replication protocol. Maybe `supported_schema_ids` as we use in the config file, or something else more self-explanatory.

Reply:
Hmm, after reading through the rest of the PR i see that announce is of course closely coupled to the rest of the replication service and protocol code. So maybe this naming is fine, interested to hear what you think though @adzialocha . I'm not interested in holding up this PR in order to find _the best_ name :rofl:

Reply:
Good point! A target set is a subset of the supported ids, it should be named differently